### PR TITLE
Automate kotlin-dsl plugin publishing

### PR DESCRIPTION
**Pull request by blindpirate**

### CODE REVIEW

File: .teamcity/src/main/kotlin/util/PublishKotlinDslPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    vcs.useAbsoluteVcs(gradleMasterVersionedSettings)
+
+    requirements {
+        requiresOs(Os.LINUX)
+    }
+
+    params {
+        param("env.JAVA_HOME", javaHome(BuildToolBuildJvm, Os.LINUX))
+        param("env.GRADLE_PUBLISH_KEY", "%plugin.portal.publish.key%")
+        param("env.GRADLE_PUBLISH_SECRET", "%plugin.portal.publish.secret%")
```
Comment (wolfs) [inline .teamcity/src/main/kotlin/util/PublishKotlinDslPlugin.kt:43]: Should this be marked as secret somehow?

**Review by wolfs — APPROVED** [review]
I left some question, though I think it looks good in general.

File: .teamcity/src/main/kotlin/util/PublishKotlinDslPlugin.kt (hunk 1 above)
Comment (blindpirate) [reply .teamcity/src/main/kotlin/util/PublishKotlinDslPlugin.kt:43]: `plugin.portal.publish.key` is already defined in Root project so I thought we can reference it directly...

File: .teamcity/src/main/kotlin/util/PublishKotlinDslPlugin.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+    steps {
+        gradleWrapper {
+            name = "Publish Kotlin DSL Plugin"
+            tasks = ":kotlin-dsl-plugins:publishPlugins"
```
[the remark this answers is not in the document]
Comment (blindpirate) [reply .teamcity/src/main/kotlin/util/PublishKotlinDslPlugin.kt:48]: Yes, right.

File: .teamcity/src/main/kotlin/util/UtilProject.kt [hunk 3]
```diff
@@ -12,6 +13,10 @@ object UtilProject : Project({
     buildType(RerunFlakyTest(Os.MACOS))
     buildType(WarmupEc2Agent)
 
+    if (VersionedSettingsBranch.fromDslContext().isMaster) {
```
[the remark this answers is not in the document]
Comment (blindpirate) [reply .teamcity/src/main/kotlin/util/UtilProject.kt:16]: I hesitated a little bit, WDYT @eskatos @bamboo ?

File: .teamcity/src/main/kotlin/util/PublishKotlinDslPlugin.kt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    name = "Publish Kotlin DSL Plugin"
+    id("Util_PublishKotlinDslPlugin")
+    uuid = "${VersionedSettingsBranch.fromDslContext().branchName.toCapitalized()}_Util_PublishKotlinDslPlugin"
+    vcs.useAbsoluteVcs(gradleMasterVersionedSettings)
+
+    requirements {
+        requiresOs(Os.LINUX)
+    }
+
+    params {
```
[the remark this answers is not in the document]
Comment (blindpirate) [reply .teamcity/src/main/kotlin/util/PublishKotlinDslPlugin.kt:40]: It's defined in https://github.com/gradle/gradle/blob/565b519897a49c587452339b0210e131f9d47e35/subprojects/kotlin-dsl-plugins/build.gradle.kts#L9

File: .teamcity/src/main/kotlin/util/UtilProject.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (eskatos) [reply .teamcity/src/main/kotlin/util/UtilProject.kt:16]: It happened in the past that we needed to publish a new version from `release`. I would not forbid it as if it comes up again it'll complicate things.

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**